src/javascripts/components/mobile-navigation.js

```diff
@@ -1,89 +1,112 @@
-  $subNavToggler: $('.js-mobile-nav-subnav-toggler'),
-  subNavActiveClass: 'app-mobile-nav__subnav--active',
-  subNavTogglerActiveClass: 'app-mobile-nav__subnav-toggler--active',
-
-  bindUIEvents: function () {
-    MobileNav.$mobileNavToggler.on('click', function (e) {
-      if (MobileNav.$mobileNav.hasClass(MobileNav.mobileNavActiveClass)) {
-        MobileNav.$mobileNav.removeClass(MobileNav.mobileNavActiveClass)
-        MobileNav.$mobileNav.attr('aria-hidden', 'true')
-
-        MobileNav.$mobileNavToggler.removeClass(MobileNav.mobileNavTogglerActiveClass)
-        MobileNav.$mobileNavToggler.attr('aria-expanded', 'false')
-      } else {
-        MobileNav.$mobileNav.addClass(MobileNav.mobileNavActiveClass)
-        MobileNav.$mobileNav.attr('aria-hidden', 'false')
-
-        MobileNav.$mobileNavToggler.attr('aria-expanded', 'true')
-        MobileNav.$mobileNavToggler.addClass(MobileNav.mobileNavTogglerActiveClass)
-      }
-    })
+import 'govuk-frontend/vendor/polyfills/Element/prototype/classList'
+
+import common from 'govuk-frontend/common'
+var nodeListForEach = common.nodeListForEach
+
+var mobileNavActiveClass = 'app-mobile-nav--active'
+var mobileNavTogglerActiveClass = 'app-header-mobile-nav-toggler--active'
+var subNavActiveClass = 'app-mobile-nav__subnav--active'
+var subNavTogglerActiveClass = 'app-mobile-nav__subnav-toggler--active'
+
+function MobileNav ($module) {
+  this.$module = $module || document
+  this.$mobileNav = this.$module.querySelector('.js-app-mobile-nav')
+  this.$mobileNavToggler = this.$module.querySelector('.js-app-mobile-nav-toggler')
+  this.$subNavTogglers = this.$module.querySelectorAll('.js-mobile-nav-subnav-toggler')
+}
+
+MobileNav.prototype.bindUIEvents = function () {
+  var $mobileNav = this.$mobileNav
+  var $mobileNavToggler = this.$mobileNavToggler
+  var $subNavTogglers = this.$subNavTogglers
+
+  $mobileNavToggler.addEventListener('click', function (e) {
+    if ($mobileNav.classList.contains(mobileNavActiveClass)) {
+      $mobileNav.classList.remove(mobileNavActiveClass)
+      $mobileNav.setAttribute('aria-hidden', 'true')
+
+      $mobileNavToggler.classList.remove(mobileNavTogglerActiveClass)
+      $mobileNavToggler.setAttribute('aria-expanded', 'false')
+    } else {
+      $mobileNav.classList.add(mobileNavActiveClass)
+      $mobileNav.setAttribute('aria-hidden', 'false')
 
-    MobileNav.$subNavToggler.on('click', function () {
-      var $toggler = $(this)
-      var $nextSubNav = $(this).next(MobileNav.$subNav)
+      $mobileNavToggler.setAttribute('aria-expanded', 'true')
+      $mobileNavToggler.classList.add(mobileNavTogglerActiveClass)
+    }
+  })
 
-      if ($nextSubNav.length > 0) {
-        if ($nextSubNav.hasClass(MobileNav.subNavActiveClass)) {
-          $nextSubNav.removeClass(MobileNav.subNavActiveClass)
-          $toggler.removeClass(MobileNav.subNavTogglerActiveClass)
+  nodeListForEach($subNavTogglers, function ($toggler) {
```

Comment:
Rather than adding multiple events to sub nav toggles individually, would it make sense to delegate this to a single click event handler at the $module level?